docs/repo.toml

```diff
@@ -371,21 +371,19 @@ doxygen_predefined = [
   "_CCCL_GLOBAL_CONSTANT=constexpr",
   "_CCCL_HOST=",
   "_CCCL_HOST_DEVICE=",
+  "_CCCL_IF_CONSTEXPR=if constexpr",
   "_CCCL_NODISCARD=[[nodiscard]]",
   "_CCCL_NODISCARD_FRIEND=",
   "_CCCL_STD_VER=2020",
+  "_CCCL_TRAIT(x, y)=x<y>::value",
   "_CUDA_VMR=cuda::mr",
   "_CUDA_VSTD=cuda::std",
   "_LIBCUDACXX_AND=&&",
   "_LIBCUDACXX_EAT_REST(x)=",
   "_LIBCUDACXX_GLOBAL_CONSTANT=inline",
   "_LIBCUDACXX_INLINE_VAR=inline",
-  "_LIBCUDACXX_REQUIRES(...)=",
-  "_LIBCUDACXX_TEMPLATE(A)=template<A>",
-  "_LIBCUDACXX_TEMPLATE(A, B)=template<A, B>",
-  "_LIBCUDACXX_TEMPLATE(A, B, C)=template<A, B, C>",
-  "_LIBCUDACXX_TEMPLATE(A, B, C, D)=template<A, B, C, D>",
-  "_LIBCUDACXX_TEMPLATE(A, B, C, D, E)=template<A, B, C, D, E>",
+  "_LIBCUDACXX_REQUIRES(x)= ::cuda::std::__enable_if_t<x, int> = 0>",
+  "_LIBCUDACXX_TEMPLATE(x)=template<x, ",
```

Comment:
do we really want our docs to have the `enable_if` gunk?

Reply:
The issue is that doxygen does not support `requires` so we either have SFINAE or nothing